hi-ml/src/health_ml/runner.py

```diff
@@ -0,0 +1,299 @@
+        assert isinstance(container, param.Parameterized)
+        parser_ = create_argparser(container)
+        # For each parser, feed in the unknown settings from the previous parser. All commandline args should
+        # be consumed by name, hence fail if there is something that is still unknown.
+        parser_result_ = parse_arguments(parser_, args=parser_result.unknown)
+        # Apply the overrides and validate. Overrides can come from either YAML settings or the commandline.
+        _ = apply_overrides(container, parser_result_.overrides)  # type: ignore
+        container.validate()
+
+        self.lightning_container = container
+
+        return parser_result_
+
+    def run(self) -> Tuple[LightningContainer, AzureRunInfo]:
+        """
+        The main entry point for training and testing models from the commandline. This chooses a model to train
+        via a commandline argument, runs training or testing, and writes all required info to disk and logs.
+
+        :return: a tuple of the LightningContainer object and an AzureRunInfo containing all information about
+            the present run (whether running in AzureML or not)
+        """
+        # Usually, when we set logging to DEBUG, we want diagnostics about the model
+        # build itself, but not the tons of debug information that AzureML submissions create.
+        logging_to_stdout(logging.INFO if is_local_rank_zero() else "ERROR")
+        initialize_rpdb()
+        self.parse_and_load_model()
+        azure_run_info = self.submit_to_azureml_if_needed()
+        self.run_in_situ(azure_run_info)
+        return self.lightning_container, azure_run_info
+
+    def submit_to_azureml_if_needed(self) -> AzureRunInfo:
+        """
+        Submit a job to AzureML, returning the resulting Run object, or exiting if we were asked to wait for
+        completion and the Run did not succeed.
+
+        :return: an AzureRunInfo object containing all of the details of the present run. If AzureML is not
+            specified, the attribute 'run' will None, but the object still contains helpful information
+            about datasets etc
+        """
+        root_folder = self.project_root
+        entry_script = Path(sys.argv[0]).resolve()
+        script_params = sys.argv[1:]
+
+        additional_conda_env_files = self.lightning_container.additional_env_files
+        additional_env_files: Optional[List[Path]]
+        if additional_conda_env_files is not None:
+            additional_env_files = [Path(f) for f in additional_conda_env_files]
+        else:
+            additional_env_files = None
+
+        conda_dependencies_files = get_all_environment_files(self.project_root,
+                                                             additional_files=additional_env_files)
+        pip_requirements_files = get_all_pip_requirements_files()
+
+        # Merge the project-specific depclass Experendencies with the packages and write unified definition
```

Comment:
```suggestion
        # Merge the project-specific dependencies with the packages and write unified definition
```

Reply:
Whoops - a case of "CTRL +SHIFT + F" not opening quickly enough!